docs/persistent_compilation_cache.md

```diff
@@ -67,8 +67,74 @@ Cloud Storage (GCS) bucket. We recommend the following configuration:
+* Compression algorithm used to compress the compiled executable
+* Any custom hooks
+
+When the signature for a function created using the parameters above matches that of a compiled function in the persistent cache, the function will not be compiled but just read and deserialized from the persistent cache. 
+
+### Compile time dependent caching
+
+JAX only caches executables that take a certain amount of time to compile. This threshold is controlled by the `jax_persistent_cache_min_compile_time_secs` configuration option. To cache every executable that is compiled regardless of compile time, set this value to zero:
+
+```python
+import jax
+
+jax.config.update("jax_persistent_cache_min_compile_time_secs", 0)
+```
+
+## Different Runtimes
+
+Below we outline some observed behavior of the persistent compilation cache in a variety of different runtimes as it relates to which processes write to the cache.
+
+### Single node with single process and single device
+
+This is the simplest runtime and the only process that compiles and writes to the compilation cache is the singular process. The number of devices does not matter to the cache key in this setup, only the type of device does. 
+
+### Single node with single process and multiple devices
+
+Once again the only process that compiles and writes to the compilation cache is the singular proess. The difference between this setup and the previous is that now the number of devices matters in addition to the type of device. 
+
+### Multiple process and multiple devices (on either single or multiple nodes)
+
+In this runtime the first time a program is run (the persistent cache is cold / empty) all processes will compile, but only the process with rank 0 in the global communication group will write to the persistent cache. In subsequent runs, all processes will attempt to read from the persistent cache, so it is important for the persistent cache to be in a shared file system (eg: NFS) or remote storage (eg: GFS). If the persistent cache is local to rank 0, then all processes except rank 0 will once again compile in subsequent runs as a result of a compilation cache miss. 
+
+## Logging cache activity
+
+It can be helpful to examine what exactly is happening with the persistent compilation cache for debugging. While there is no singular canonical way of debugging and examining what's happening in the compilation cache, here are a few suggestions on how to begin.
+
+### Examining cache misses
+
+To merely examine and understand why there are cache misses JAX includes a configuration flag that enables the logging of all cache misses (including persistent compilation cache misses) with their explanations. This can be enabled by setting the following configuration.
+
+```python
+import jax
+
+jax.config.update("jax_explain_cache_misses", True)
+```
```

Comment:
I think `jax_explain_cache_misses` is still WIP for explaining persistent compilation cache miss? (See my design doc above)